mobile/fula.go

```diff
@@ -48,66 +64,184 @@ type DirEntry struct {
 }
 
 type Fula struct {
-	Node  rhost.RoutedHost
+	h     host.Host
 	ctx   context.Context
 	mybox []peer.AddrInfo
+	ls    ipld.LinkSystem
+	gx    graphsync.GraphExchange
+	ds    datastore.Batching
 }
 
 func NewFula(repoPath string) (*Fula, error) {
-	err := logging.SetLogLevelRegex(".*", "DEBUG")
-
-	if err != nil {
-		// TODO: don't panic!
-		panic("logger failed")
-	}
 	ctx := context.Background()
-	node, err := create(ctx, repoPath)
+	host, err := create(ctx, repoPath)
 	if err != nil {
 		return nil, err
 	}
-	f := &Fula{Node: *node, ctx: ctx}
+	f := &Fula{h: host, ctx: ctx}
```

Comment:
Out of the scope for this PR but something to fix later: Avoid keeping a reference to `ctx` unless we really have to. Instesd, separate the operations that needed into functions like `Start` and `Shutdown` and take context on those functions.